vaas/vaas/vcl/renderer.py

```diff
@@ -167,10 +167,24 @@ def __init__(self, varnish, input_data):
             'probe': self.prepare_probe(vcl_directors),
             'active_director': active_directors,
             'routes': self.prepare_route(self.varnish, cluster_directors),
+            'redirects': self.prepare_redirect(),
             'cluster_directors': cluster_directors,
             'mesh_routing': varnish.cluster.service_mesh_routing
         }
 
+    @collect_processing
+    def prepare_redirect(self):
+        redirects = {}
+        for redirect in self.input.redirects:
+            if redirect.src_domain in self.varnish.cluster.domainmapping_set.all():
+                records = redirects.get(redirect.src_domain.domain, [])
+                if records:
+                    records.append(redirect)
+                else:
+                    redirects[redirect.src_domain.domain] = [redirect]
```

Comment:
It can be simplified by using the walrus operator:
```
if entries := redirects.get(redirect.src_domain.domain, []):
    entries.append(redirect)
else:
    redirects[redirect.src_domain.domain] = [redirect]
```